contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -2201,50 +2209,101 @@ gen_sp_droprolemember_subcmds(const char *user, const char *member)
 	return res;
 }
 
-/*
- * Helper function to execute a utility command using
- * ProcessUtility(). Caller should make sure their
- * inputs are sanitized to prevent unexpected behaviour.
- */
-static void
-exec_utility_cmd_helper(char *query_str)
+static void 
+update_bbf_server_options(char *servername, char *querytimeout, bool isInsert)
 {
-	List	   *parsetree_list;
-	Node	   *stmt;
-	PlannedStmt *wrapper;
-
-	parsetree_list = raw_parser(query_str, RAW_PARSE_DEFAULT);
+	Relation	bbf_servers_def_rel;
+	TupleDesc	bbf_servers_def_rel_dsc;
+	Datum		new_record[BBF_SERVERS_DEF_NUM_COLS];
+	bool		new_record_nulls[BBF_SERVERS_DEF_NUM_COLS];
+	bool		new_record_repl[BBF_SERVERS_DEF_NUM_COLS];
+	ScanKeyData		key;
+	HeapTuple		tuple, old_tuple;
+	TableScanDesc	tblscan;
+	int32		query_timeout;
+
+	if (querytimeout == NULL)
+		query_timeout = 0;
+	else
+		query_timeout = atoi(querytimeout);
 
-	if (list_length(parsetree_list) != 1)
+	if (query_timeout < 0 || strspn(querytimeout, "0123456789") != strlen(querytimeout))
```

Comment:
this if should be part of the previous else block. Also, we can move the strspn() check before atoi(). atoi() will always get a valid whole number as argument in that case